src/sentry/sentry_metrics/configuration.py

```diff
@@ -41,23 +36,32 @@ def _register_ingest_config(config: MetricsIngestConfiguration) -> None:
     _METRICS_INGEST_CONFIG_BY_USE_CASE[config.use_case_id] = config
 
 
-def get_ingest_config(use_case_key: UseCaseKey) -> MetricsIngestConfiguration:
+def get_ingest_config(use_case_key: UseCaseKey, db_backend: str) -> MetricsIngestConfiguration:
     if len(_METRICS_INGEST_CONFIG_BY_USE_CASE) == 0:
         _register_ingest_config(
             MetricsIngestConfiguration(
-                db_model=DbKey.STRING_INDEXER,
+                db_backend=db_backend,
```

Comment:
This is going to cause an issue. 
The `get_ingest_config` is called in multiple places within sentry and the indexer to get a specific indexer config. When you run this method from the indexer code you will always pass the same use case key so no problem. But sentry (being it a monolith) needs to have access to all indexer configs at the same time. So, if you use `db_backend` to initialize the config, you will fundamentally be ignoring that parameter for all subsequent calls. 

If you call `get_ingest_config('release_health', 'spanner')` you get the spanner indexer.
If you then call `get_ingest_config('release_health', 'postgres')`, the db_backend is ignored and the spanner one is returned as the first called has initialized the config. 

Basically you cannot use the parameters passed to the method in the initialization of the config as that is then stored in the module and not changed anymore no matter which parameters you pass.

Reply:
ah yeah okay cause `_METRICS_INGEST_CONFIG_BY_USE_CASE[use_case_key]` is using the use_case_key for look up. makes sense.